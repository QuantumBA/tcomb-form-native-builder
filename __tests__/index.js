import {mount}   from 'enzyme'
import React     from 'react'

import Builder from '../src'


describe('load definition', function()
{
  it('no definition', function()
  {
    const wrapper = mount(<Builder/>)

    expect(wrapper).toMatchSnapshot()
  })
})
