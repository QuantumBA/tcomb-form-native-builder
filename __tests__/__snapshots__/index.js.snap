// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load definition no definition 1`] = `
<Builder>
  <Form
    i18n={
      Object {
        "add": "Add",
        "down": "↓",
        "optional": " (optional)",
        "remove": "✘",
        "required": "",
        "up": "↑",
      }
    }
    stylesheet={
      Object {
        "button": Object {
          "alignSelf": "stretch",
          "backgroundColor": "#48BBEC",
          "borderColor": "#48BBEC",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 36,
          "justifyContent": "center",
          "marginBottom": 10,
        },
        "buttonText": Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 18,
        },
        "checkbox": Object {
          "error": Object {
            "marginBottom": 4,
          },
          "normal": Object {
            "marginBottom": 4,
          },
        },
        "controlLabel": Object {
          "error": Object {
            "color": "#a94442",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          },
          "normal": Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          },
        },
        "dateTouchable": Object {
          "error": Object {},
          "normal": Object {},
        },
        "dateValue": Object {
          "error": Object {
            "color": "#a94442",
            "fontSize": 17,
            "marginBottom": 5,
            "padding": 7,
          },
          "normal": Object {
            "color": "#000000",
            "fontSize": 17,
            "marginBottom": 5,
            "padding": 7,
          },
        },
        "datepicker": Object {
          "error": Object {
            "marginBottom": 4,
          },
          "normal": Object {
            "marginBottom": 4,
          },
        },
        "errorBlock": Object {
          "color": "#a94442",
          "fontSize": 17,
          "marginBottom": 2,
        },
        "fieldset": Object {},
        "formGroup": Object {
          "error": Object {
            "marginBottom": 10,
          },
          "normal": Object {
            "marginBottom": 10,
          },
        },
        "helpBlock": Object {
          "error": Object {
            "color": "#999999",
            "fontSize": 17,
            "marginBottom": 2,
          },
          "normal": Object {
            "color": "#999999",
            "fontSize": 17,
            "marginBottom": 2,
          },
        },
        "pickerContainer": Object {
          "error": Object {
            "borderColor": "#a94442",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginBottom": 4,
          },
          "normal": Object {
            "borderColor": "#cccccc",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginBottom": 4,
          },
          "open": Object {},
        },
        "pickerTouchable": Object {
          "active": Object {
            "borderBottomWidth": 1,
            "borderColor": "#cccccc",
          },
          "error": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
          },
          "normal": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
          },
        },
        "pickerValue": Object {
          "error": Object {
            "fontSize": 17,
            "paddingLeft": 7,
          },
          "normal": Object {
            "fontSize": 17,
            "paddingLeft": 7,
          },
        },
        "select": Object {
          "error": Object {},
          "normal": Object {},
        },
        "textbox": Object {
          "error": Object {
            "borderColor": "#a94442",
            "borderRadius": 4,
            "borderWidth": 1,
            "color": "#000000",
            "fontSize": 17,
            "height": 36,
            "marginBottom": 5,
            "paddingHorizontal": 7,
            "paddingVertical": 7,
          },
          "normal": Object {
            "borderColor": "#cccccc",
            "borderRadius": 4,
            "borderWidth": 1,
            "color": "#000000",
            "fontSize": 17,
            "height": 36,
            "marginBottom": 5,
            "paddingHorizontal": 7,
            "paddingVertical": 7,
          },
          "notEditable": Object {
            "backgroundColor": "#eeeeee",
            "borderColor": "#cccccc",
            "borderRadius": 4,
            "borderWidth": 1,
            "color": "#777777",
            "fontSize": 17,
            "height": 36,
            "marginBottom": 5,
            "paddingHorizontal": 7,
            "paddingVertical": 7,
          },
        },
        "textboxView": Object {
          "error": Object {},
          "normal": Object {},
          "notEditable": Object {},
        },
      }
    }
    templates={
      Object {
        "checkbox": [Function],
        "datepicker": [Function],
        "list": [Function],
        "select": [Function],
        "struct": [Function],
        "textbox": [Function],
      }
    }
    type={[Function]}
  >
    <Textbox
      ctx={
        Object {
          "auto": "labels",
          "context": undefined,
          "i18n": Object {
            "add": "Add",
            "down": "↓",
            "optional": " (optional)",
            "remove": "✘",
            "required": "",
            "up": "↑",
          },
          "path": Array [],
          "stylesheet": Object {
            "button": Object {
              "alignSelf": "stretch",
              "backgroundColor": "#48BBEC",
              "borderColor": "#48BBEC",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 36,
              "justifyContent": "center",
              "marginBottom": 10,
            },
            "buttonText": Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 18,
            },
            "checkbox": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "controlLabel": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
            },
            "dateTouchable": Object {
              "error": Object {},
              "normal": Object {},
            },
            "dateValue": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
            },
            "datepicker": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "errorBlock": Object {
              "color": "#a94442",
              "fontSize": 17,
              "marginBottom": 2,
            },
            "fieldset": Object {},
            "formGroup": Object {
              "error": Object {
                "marginBottom": 10,
              },
              "normal": Object {
                "marginBottom": 10,
              },
            },
            "helpBlock": Object {
              "error": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
              "normal": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
            },
            "pickerContainer": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "open": Object {},
            },
            "pickerTouchable": Object {
              "active": Object {
                "borderBottomWidth": 1,
                "borderColor": "#cccccc",
              },
              "error": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
              "normal": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
            },
            "pickerValue": Object {
              "error": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
              "normal": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
            },
            "select": Object {
              "error": Object {},
              "normal": Object {},
            },
            "textbox": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "notEditable": Object {
                "backgroundColor": "#eeeeee",
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#777777",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
            },
            "textboxView": Object {
              "error": Object {},
              "normal": Object {},
              "notEditable": Object {},
            },
          },
          "templates": Object {
            "checkbox": [Function],
            "datepicker": [Function],
            "list": [Function],
            "select": [Function],
            "struct": [Function],
            "textbox": [Function],
          },
          "uidGenerator": UIDGenerator {
            "counter": 0,
            "seed": "tfid-0-",
          },
        }
      }
      onChange={[Function]}
      options={Object {}}
      type={[Function]}
    >
      <Component
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <react-native-mock>
          <Component
            style={Object {}}
          >
            <react-native-mock>
              <Component
                onChangeText={[Function]}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              >
                <react-native-mock />
              </Component>
            </react-native-mock>
          </Component>
        </react-native-mock>
      </Component>
    </Textbox>
  </Form>
</Builder>
`;
